### PR TITLE
Disallow empty URLs

### PR DESCRIPTION
**Pull request by dimaryaz**
Defaulting to "." is counter-intuitive. If you really want ".", then use ".".

### CODE REVIEW

File: api/python/t4/api.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def list_packages(registry=None):
         A list of strings containing the names of the packages        
     """
     if not registry:
-        registry = ''
+        registry = 'whatever'
```
Comment (dimaryaz) [inline api/python/t4/api.py]: Obvious question: what _should_ this be? Probably not the current dir. `get_local_package_registry()`? We need to refactor some of that code.

Comment (stevededalus) [reply api/python/t4/api.py]: why not the default local one?

File: api/python/t4/util.py [hunk 2]
```diff
@@ -76,6 +76,9 @@ def split_path(path, require_subpath=False):
 
 def fix_url(url):
     """Convert non-URL paths to file:// URLs"""
+    if not url:
```
Comment (stevededalus) [inline api/python/t4/util.py]: Don't we want an empty one to resolve to the directory at the current location

File: api/python/t4/api.py [hunk 3]
```diff
@@ -141,7 +141,7 @@ def list_packages(registry=None):
         A list of strings containing the names of the packages        
     """
     if not registry:
-        registry = ''
```
Comment (stevededalus) [inline api/python/t4/api.py]: What's the goal of this PR?

**Comment by dimaryaz** [comment]
Fixing this in #95 instead

As for the goal: just sanity; empty URLs are an easy way to introduce bugs

**Pull request closed without merging**